dataactvalidator/validation_handlers/validationManager.py

```diff
@@ -310,55 +311,44 @@ def runValidation(self, jobId, interfaces):
-        rules = interfaces.validationDb.session.query(RuleSql).filter(RuleSql.rule_cross_file_flag == True).all()
-        failures.extend(Validator.crossValidateSql(rules, submissionId))
         bucketName = CONFIG_BROKER['aws_bucket']
         regionName = CONFIG_BROKER['aws_region']
-        errorDb = interfaces.errorDb
 
-        # Get list of file types
-        fileTypes = interfaces.validationDb.getFileTypeList()
-        # Create dict mapping file types to empty lists
-        fileTypeDict = {targetFile:[] for targetFile in fileTypes}
-
-        # Sort failures by source and target file
-        failureDict = {sourceFile:copy.deepcopy(fileTypeDict) for sourceFile in fileTypes}
-
-        for failure in failures:
-            errorDb.recordRowError(jobId, "cross_file",
-                    failure[0], failure[3], failure[5], failure[6], failure[7], failure[8])
-            # Put failure into list for this source and target file
-            sourceType = interfaces.validationDb.getFileTypeById(failure[7])
-            targetType = interfaces.validationDb.getFileTypeById(failure[8])
-            if failure[8] < failure[7]:
-                # Lower ID goes first
-                failureDict[targetType][sourceType].append(failure)
-            elif failure[8] == failure[7]:
-                raise ValueError("Both files in cross file are the same type")
-            else:
-                failureDict[sourceType][targetType].append(failure)
-
-        # For each pair of files create an error report
-        for sourceFile in fileTypes:
-            for targetFile in fileTypes:
-                sourceId = interfaces.validationDb.getFileTypeIdByName(sourceFile)
-                targetId = interfaces.validationDb.getFileTypeIdByName(targetFile)
-                if targetId <= sourceId:
-                    # Only create files in one direction
-                    continue
-                # Get filename for this error report
-                reportFilename = self.getFileName(interfaces.errorDb.getCrossReportName(submissionId, sourceFile,targetFile))
-                with self.getWriter(regionName, bucketName, reportFilename,
+
+        # use db to get a list of the cross-file combinations
+        targetFiles = validationDb.session.query(FileType).subquery()
+        crossFileCombos = validationDb.session.query(
+            FileType.name.label('source_file_name'),
+            FileType.file_id.label('source_file_id'),
+            targetFiles.c.name.label('target_file_name'),
+            targetFiles.c.file_id.label('target_file_id')
+        ).filter(FileType.file_id < targetFiles.c.file_id)
+
+        # get all cross file rules from db
+        crossFileRules = validationDb.session.query(RuleSql).filter(RuleSql.rule_cross_file_flag==True)
+
+        # for each cross-file combo, run associated rules and create error report
+        for row in crossFileCombos:
+            comboRules = crossFileRules.filter(
```

Comment:
Is this filter sufficient for making sure we include the ["two way rules" you explained](https://github.com/fedspendingtransparency/data-act-broker-backend/pull/117#discussion_r69129088). I was playing around with this query before I understood that particular use case, so want to make sure we're covering all the bases.